### ISSUE TITLE
JEV-FB-01: Fallback - Structured-output LLM routing

### ISSUE THREAD

**Issue by jmlusu** [jmlusu/light-speed-holdings#347]
## Task
Implement structured-output LLM routing as fallback if Jev validation fails.

## Approach
Use JSON mode + Instructor library for constrained LLM outputs.

## Implementation
```python
from instructor import patch
from pydantic import BaseModel

class RoutingDecision(BaseModel):
    intent: Literal["research", "coding", "creative", "ops"]
    complexity: Literal["trivial", "standard", "expert", "novel"]
    risk: Literal["low", "medium", "high", "critical"]
    requires_human: bool
    confidence: float

client = patch(openai_client)
decision = client.chat.completions.create(
    model="gpt-5.6-terra",
    response_model=RoutingDecision,
    messages=[{"role": "user", "content": prompt}],
    temperature=0.1
)
```

## Calibration Layer
- Temperature scaling on logits for confidence calibration
- Platt scaling on validation set
- Target ECE < 0.15 (vs Jev's < 0.1)

## Acceptance Criteria
- [ ] Structured output parsing 100% reliable
- [ ] Calibration ECE < 0.15 on validation set
- [ ] Latency < 2s p99 (vs Jev 0.5s)
- [ ] Cost < $0.01/task (vs Jev $0.0004)
- [ ] Falls back gracefully on parse failures

## Trigger
- JEV-010 decision = NO-GO
